code/modules/reagents/Chemistry-Reagents/Pyrotechnic-Reagents.dm

```diff
@@ -200,3 +200,51 @@
 		holder.chem_temp += 10
 		holder.handle_reactions()
 	..()
+
+
+/datum/reagent/cryogenic_fluid
+	name = "Cryogenic Fluid"
+	id = "cryogenic_fluid"
+	description = "Extremely cold superfluid used to put out fires that can freeze people solid, chills people when ingested and has a volume dependant freeze reaction on touch which worsens existing brute damage."
+	color = "#b3ffff" // rgb: 0, 255, 255
+	metabolization_rate = 2
+
+/datum/reagent/cryogenic_fluid/on_tick()
+	holder.chem_temp -= 5
+	..()
+/datum/reagent/cryogenic_fluid/reaction_mob(mob/living/M, method=TOUCH, reac_volume, show_message = 1)
+	if(iscarbon(M) && M.stat != DEAD)
+		if(method in list(INGEST,INJECT))
+			M.adjust_fire_stacks(-(reac_volume))
+			M.adjustStaminaLoss(reac_volume)
+			M.adjustFireLoss(reac_volume)
+			M.bodytemperature = max(M.bodytemperature - 50, TCMB)
+			if(show_message)
+				M << "<span class='warning'>You feel like you are freezing from the inside!</span>"
+		else
+			if(show_message)
```

Comment:
Wouldn't this make more sense to put this inside the if(reac_volume >= 5) since you don't actually get colder unless that's true?